ios/ReactNativeShareExtension.m

```diff
@@ -196,6 +196,37 @@ - (void)processShareAttachment:(NSEnumerator<NSItemProvider*> *)attachmentEnumer
           @try {
             NSURL *url = (NSURL *)item;
             [dataArray addObject:[url absoluteString]];
+            [self processShareAttachment:attachmentEnumerator intoArray:dataArray withCompletionHandler:handler];
+          } @catch (NSException *exception) {
+            if (handler) {
+              handler(nil, exception);
+            }
+          }
+        }];
+    } else if ([provider hasItemConformingToTypeIdentifier:(NSString *)kUTTypePropertyList]) {
+        [provider loadItemForTypeIdentifier:(NSString *)kUTTypePropertyList options:nil completionHandler:^(NSDictionary *item, NSError *error) {
+
+          @try {
+              NSDictionary *results = (NSDictionary *)item;
+              NSMutableDictionary *extensionData = [[results objectForKey:NSExtensionJavaScriptPreprocessingResultsKey] objectForKey:@"extension_java_script_data"];
+
+              if (extensionData) {
+                  NSString *pageUrl = extensionData[@"url"];
```

Comment:
What happens if this is undefined / nil?

Reply:
url will be undefined/nil ? but i'm not sure if it's possible because we always pass it here https://github.com/UpnextHq/upnext-apps-mobile/pull/723/files#diff-b9d20143eaff20ea8ed1f2a948da030a6af6e78e7ff08a4dd24eb022ac120225R16